presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPartitionManager.java

```diff
@@ -78,6 +91,88 @@ public List<String> getEffectivePartitions(
                 partitionPredicate.put(partitionColumn, Domain.all(column.getHiveType().getType(typeManager)));
             }
         }
-        return metastore.getPartitionNamesByFilter(metastoreContext, schemaName, tableName, partitionPredicate);
+
+        // fetch partition names from metastore
+        List<String> partitionNames = metastore.getPartitionNamesByFilter(metastoreContext, schemaName, tableName, partitionPredicate);
+        List<Type> partitionTypes = partitionColumns.stream()
+                .map(column -> column.getType().getType(typeManager))
+                .collect(toList());
+        Constraint<ColumnHandle> constraint = new Constraint<>(tupleDomain);
+        // match partition and get partition only we need
+        Iterable<HudiPartition> partitionsIterable = () -> partitionNames.stream()
```

Comment:
nits: partionsIterable -> remainingPartitions, for better readability

Reply:
And  It's unnecessary to create a `HudiPartition`, since we actually only need the names of the remaining partitions.